File: include/extras-kconfig/libkemailsettings.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const kemailsettings_enums = enums;
const std = @import("std");

/// https://api-staging.kde.org/kemailsettings.html
pub const kemailsettings = struct {
    /// New constructs a new KEMailSettings object.
    ///
    ///
    pub fn New() QtC.KEMailSettings {
        return qtc.KEMailSettings_new();
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qobject.html#tr)
    ///
    /// ``` sourceText: []const u8, allocator: std.mem.Allocator ```
    pub fn Tr(sourceText: []const u8, allocator: std.mem.Allocator) []const u8 {
        const sourceText_Cstring = sourceText.ptr;
        const _str = qtc.KEMailSettings_Tr(sourceText_Cstring);
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kemailsettings.Tr: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api-staging.kde.org/kemailsettings.html#profiles)
    ///
    /// ``` self: QtC.KEMailSettings, allocator: std.mem.Allocator ```
    pub fn Profiles(self: ?*anyopaque, allocator: std.mem.Allocator) [][]const u8 {
        const _arr: qtc.libqt_list = qtc.KEMailSettings_Profiles(@ptrCast(self));
        const _str: [*]qtc.libqt_string = @ptrCast(@alignCast(_arr.data));
        defer {
            for (0.._arr.len) |i| {
                qtc.libqt_string_free(@ptrCast(&_str[i]));
            }
            qtc.libqt_free(_arr.data);
        }
        const _ret = allocator.alloc([]const u8, _arr.len) catch @panic("kemailsettings.Profiles: Memory allocation failed");
        for (0.._arr.len) |i| {
            const _data = _str[i];
            const _buf = allocator.alloc(u8, _data.len) catch @panic("kemailsettings.Profiles: Memory allocation failed");
            @memcpy(_buf, _data.data[0.._data.len]);
            _ret[i] = _buf;
        }
        return _ret;
    }

    /// [Qt documentation](https://api-staging.kde.org/kemailsettings.html#setProfile)
    ///
    /// ``` self: QtC.KEMailSettings, s: []const u8 ```
    pub fn SetProfile(self: ?*anyopaque, s: []const u8) void {
        const s_str = qtc.libqt_string{
            .len = s.len,
            .data = s.ptr,
        };
        qtc.KEMailSettings_SetProfile(@ptrCast(self), s_str);
    }

    /// [Qt documentation](https://api-staging.kde.org/kemailsettings.html#defaultProfileName)
    ///
    /// ``` self: QtC.KEMailSettings, allocator: std.mem.Allocator ```
    pub fn DefaultProfileName(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KEMailSettings_DefaultProfileName(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kemailsettings.DefaultProfileName: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api-staging.kde.org/kemailsettings.html#setDefault)
    ///
    /// ``` self: QtC.KEMailSettings, def: []const u8 ```
    pub fn SetDefault(self: ?*anyopaque, def: []const u8) void {
        const def_str = qtc.libqt_string{
            .len = def.len,
            .data = def.ptr,
        };
        qtc.KEMailSettings_SetDefault(@ptrCast(self), def_str);
    }

    /// [Qt documentation](https://api-staging.kde.org/kemailsettings.html#getSetting)
    ///
    /// ``` self: QtC.KEMailSettings, s: kemailsettings_enums.Setting, allocator: std.mem.Allocator ```
    pub fn GetSetting(self: ?*anyopaque, s: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KEMailSettings_GetSetting(@ptrCast(self), @intCast(s));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kemailsettings.GetSetting: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api-staging.kde.org/kemailsettings.html#setSetting)
    ///
    /// ``` self: QtC.KEMailSettings, s: kemailsettings_enums.Setting, v: []const u8 ```
    pub fn SetSetting(self: ?*anyopaque, s: i32, v: []const u8) void {
        const v_str = qtc.libqt_string{
            .len = v.len,
            .data = v.ptr,
        };
        qtc.KEMailSettings_SetSetting(@ptrCast(self), @intCast(s), v_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qobject.html#tr)
    ///
    /// ``` sourceText: []const u8, disambiguation: []const u8, allocator: std.mem.Allocator ```
    pub fn Tr2(sourceText: []const u8, disambiguation: []const u8, allocator: std.mem.Allocator) []const u8 {
        const sourceText_Cstring = sourceText.ptr;
        const disambiguation_Cstring = disambiguation.ptr;
        const _str = qtc.KEMailSettings_Tr2(sourceText_Cstring, disambiguation_Cstring);
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kemailsettings.Tr2: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qobject.html#tr)
    ///
    /// ``` sourceText: []const u8, disambiguation: []const u8, n: i32, allocator: std.mem.Allocator ```
    pub fn Tr3(sourceText: []const u8, disambiguation: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const sourceText_Cstring = sourceText.ptr;
        const disambiguation_Cstring = disambiguation.ptr;
        const _str = qtc.KEMailSettings_Tr3(sourceText_Cstring, disambiguation_Cstring, @intCast(n));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kemailsettings.Tr3: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api-staging.kde.org/kemailsettings.html#dtor.KEMailSettings)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KEMailSettings ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KEMailSettings_Delete(@ptrCast(self));
    }
};

/// https://api-staging.kde.org/kemailsettings.html#types
pub const enums = struct {
    pub const Setting = enum {
        pub const ClientProgram: i32 = 0;
        pub const ClientTerminal: i32 = 1;
        pub const RealName: i32 = 2;
        pub const EmailAddress: i32 = 3;
        pub const ReplyToAddress: i32 = 4;
        pub const Organization: i32 = 5;
    };
};

File: src_original/extras-kconfig/libkemailsettings.cpp
#include <KEMailSettings>
#include <QList>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <kemailsettings.h>
#include "libkemailsettings.h"
#include "libkemailsettings.hxx"

KEMailSettings* KEMailSettings_new() {
    return new KEMailSettings();
}

libqt_string KEMailSettings_Tr(const char* sourceText) {
    QString _ret = KEMailSettings::tr(sourceText);
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_list /* of libqt_string */ KEMailSettings_Profiles(const KEMailSettings* self) {
    QList<QString> _ret = self->profiles();
    // Convert QList<> from C++ memory to manually-managed C memory
    libqt_string* _arr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * (_ret.size() + 1)));
    for (qsizetype i = 0; i < _ret.size(); ++i) {
        QString _lv_ret = _ret[i];
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _lv_b = _lv_ret.toUtf8();
        libqt_string _lv_str;
        _lv_str.len = _lv_b.length();
        _lv_str.data = static_cast<const char*>(malloc(_lv_str.len + 1));
        memcpy((void*)_lv_str.data, _lv_b.data(), _lv_str.len);
        ((char*)_lv_str.data)[_lv_str.len] = '\0';
        _arr[i] = _lv_str;
    }
    libqt_list _out;
    _out.len = _ret.size();
    _out.data = static_cast<void*>(_arr);
    return _out;
}

void KEMailSettings_SetProfile(KEMailSettings* self, const libqt_string s) {
    QString s_QString = QString::fromUtf8(s.data, s.len);
    self->setProfile(s_QString);
}

libqt_string KEMailSettings_DefaultProfileName(const KEMailSettings* self) {
    QString _ret = self->defaultProfileName();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void KEMailSettings_SetDefault(KEMailSettings* self, const libqt_string def) {
    QString def_QString = QString::fromUtf8(def.data, def.len);
    self->setDefault(def_QString);
}

libqt_string KEMailSettings_GetSetting(const KEMailSettings* self, int s) {
    QString _ret = self->getSetting(static_cast<KEMailSettings::Setting>(s));
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void KEMailSettings_SetSetting(KEMailSettings* self, int s, const libqt_string v) {
    QString v_QString = QString::fromUtf8(v.data, v.len);
    self->setSetting(static_cast<KEMailSettings::Setting>(s), v_QString);
}

libqt_string KEMailSettings_Tr2(const char* sourceText, const char* disambiguation) {
    QString _ret = KEMailSettings::tr(sourceText, disambiguation);
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KEMailSettings_Tr3(const char* sourceText, const char* disambiguation, int n) {
    QString _ret = KEMailSettings::tr(sourceText, disambiguation, static_cast<int>(n));
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void KEMailSettings_Delete(KEMailSettings* self) {
    delete self;
}
